cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -3336,6 +3336,26 @@ TEST(TestArrowReadWrite, NonUniqueDictionaryValues) {
   }
 }
 
+TEST(TestArrowReadWrite, DictionaryIndexBitwidthRoundtrip) {
+  auto dict_type = ::arrow::dictionary(::arrow::int8(), ::arrow::utf8());
```

Comment:
Let's perhaps be nifty and check all possible index types? Something like:
```suggestion
  for (auto index_type : {int8(), int16(), int32(), int64()}) {
    ARROW_SCOPED_TRACE("index_type = " , *index_type);
    auto dict_type = ::arrow::dictionary(index_type, ::arrow::utf8());
    ...
```